src/cmap/command_monitoring_events.ts

```diff
@@ -83,6 +95,11 @@ export class CommandSucceededEvent {
     this.commandName = commandName;
     this.duration = calculateDurationInMs(started);
     this.reply = maybeRedact(commandName, cmd, extractReply(command, reply));
+    this.serviceId = serviceId;
+  }
+
+  get hasServiceId(): boolean {
```

Comment:
The reason for adding these was for the unified tests. They are asserting on each event `hasServiceId` and the runner basically matches based on checking for the property generically. Maybe I just add `/* @internal */` so the API extractor removes them?

Reply:
Yeah I see that we have this written in the tests:
```
expect(actualEvent.hasServiceId).to.equal(expectedEvent.connectionClosedEvent.hasServiceId);
```
could they just be written like:
```
if (expectedEvent.connectionClosedEvent?.hasServiceId) {
  expect(actualEvent.serviceId).to.exist;
}
```
?

This way we wouldn't need to include the getter in the public API

Reply:
That works for the 2 cmap events but doesn't work for the command monitoring events which go through `resultCheck` (https://github.com/mongodb/node-mongodb-native/blob/4.0/test/functional/unified-spec-runner/match.ts#L109) in the unified runner and I'm hesitant to start adding more special cases into that method at this point. I feel like marking them as internal so they don't show up in the API docs as sufficient for now.

Reply:
The spec states that "driver's MAY choose how they expose the serviceId" which means there could've been any number of ways to attach this information,  I feel like a special case is warranted here because it is the case that it can be in any number of forms. [It looks like Java put this logic in their test runner](https://github.com/mongodb/mongo-java-driver/blob/master/driver-sync/src/test/functional/com/mongodb/client/unified/EventMatcher.java#L61-L69)

That being said it seems like this could've been covered by an `{ $$exists: true }` test if the spec mandated a `serviceId` property, maybe we should file a Drivers ticket to follow up on that?